### PR TITLE
op-node: remove unused AddSafeAttributes function

### PR DESCRIPTION
**Pull request by s7v7nislands**
**Description**

AddSafeAttributes is not used, remove it, and simple the logic.

### CODE REVIEW

**Review by trianglesphere — COMMENTED** [review]
Nice find. We've got some changes in https://github.com/ethereum-optimism/optimism/pull/5047 which might cause some minor conflicts. I'd like to get the merged first & then this can go in after.

**Comment by s7v7nislands** [comment]
> 

Ok, I will update the patch after #5047 merged

**Comment by s7v7nislands** [comment]
@trianglesphere should We also merge #5066 first?

**Comment by trianglesphere** [comment]
> should We also merge https://github.com/ethereum-optimism/optimism/pull/5066 first?

sure.

File: op-node/rollup/derive/engine_queue.go [hunk 1]
```diff
@@ -212,27 +207,26 @@ func (eq *EngineQueue) Step(ctx context.Context) error {
...
+		return err
+	}
+	eq.origin = newOrigin
+	eq.postProcessSafeL2() // make sure we track the last L2 safe head for every new L1 block
+	if next, err := eq.prev.NextAttributes(ctx, eq.safeHead); err == io.EOF {
+		outOfData = true
+	} else if err != nil {
+		return err
+	} else {
+		eq.safeAttributes = next
```
Comment (trianglesphere) [inline op-node/rollup/derive/engine_queue.go:226]: Could you add the `Adding next safe attributes` log (probably just `Storing safe attributes`) here?

Comment (s7v7nislands) [reply op-node/rollup/derive/engine_queue.go:226]: updated

**Review by trianglesphere — APPROVED** [review]
tyvm

**Comment by trianglesphere** [comment]
Lint failed at `rollup/derive/engine_queue.go:107`

**Pull request merged**